app/src/main/java/com/insalyon/les24heures/utils/TimeLocationSortComparator.java

```diff
@@ -0,0 +1,57 @@
+package com.insalyon.les24heures.utils;
+
+import android.location.Location;
+
+import com.insalyon.les24heures.model.DayResource;
+
+import java.util.Comparator;
+import java.util.Date;
+
+
+import static java.util.Calendar.DAY_OF_WEEK;
+
+/**
+ * Created by nicolas on 19/03/15.
+ */
+public class TimeLocationSortComparator implements Comparator<DayResource> {
+    @Override
+    public int compare(DayResource res1, DayResource res2) {
+        int ret=1;
+
+        Date start1 = res1.getSchedules().get(0).getStart();
+        Date start2 = res2.getSchedules().get(0).getStart();
+
+        if (start1.after(start2)){
+            ret = 1;
+        }
+        else if (start1.before(start2)){
+            ret = -1;
+        }
+        //If Schedules begin at the same time, we use distance to compare
+        else if(start1.equals(start2)) {
+            Location loc1 = new Location("loc");
+            loc1.setLongitude(res1.getLoc().longitude);
+            loc1.setLatitude(res1.getLoc().latitude);
+
+            Location loc2 = new Location("loc");
+            loc2.setLongitude(res2.getLoc().longitude);
+            loc2.setLatitude(res2.getLoc().latitude);
+
+            //TODO : Use real position instead of constant lastKnownPosition
+            Location lastKnownPosition = new Location("lastKnownPosition");
+            lastKnownPosition.setLatitude(45.78401554);
+            lastKnownPosition.setLongitude(4.8754406);
+
+            Integer distance1 = Math.round(lastKnownPosition.distanceTo(loc1));
+            Integer distance2 = Math.round(lastKnownPosition.distanceTo(loc2));
+
+            if (distance1 > distance2) {
```

Comment:
Les autres changements sont OK mais je ne sais pas comment passer
LastKnowPosition de l'Adapter au Comparator. Je ne peux pas rajouter un
paramètre à la méthode compare ni à l'appel de méthode sort dans Adapter.

*Nicolas Nadisic*
Responsable Logistique

*nicolas.nadisic@gmail.com <nicolas.nadisic@gmail.com>*
06 45 29 97 20
 [image: 24Heures]
20, avenue Albert Einstein
69621 Villeurbanne Cedex
04 72 43 70 00
*www.24heures.org <http://www.24heures.org>*



2015-03-20 10:52 GMT+01:00 Remi Pichon <notifications@github.com>:

> In
> app/src/main/java/com/insalyon/les24heures/utils/TimeLocationSortComparator.java
> <https://github.com/24HeuresINSA/24h-android-app/pull/35#discussion_r26828219>
> :
>
> > +            loc1.setLongitude(res1.getLoc().longitude);
> > +            loc1.setLatitude(res1.getLoc().latitude);
> > +
> > +            Location loc2 = new Location("loc");
> > +            loc2.setLongitude(res2.getLoc().longitude);
> > +            loc2.setLatitude(res2.getLoc().latitude);
> > +
> > +            //TODO : Use real position instead of constant lastKnownPosition
> > +            Location lastKnownPosition = new Location("lastKnownPosition");
> > +            lastKnownPosition.setLatitude(45.78401554);
> > +            lastKnownPosition.setLongitude(4.8754406);
> > +
> > +            Integer distance1 = Math.round(lastKnownPosition.distanceTo(loc1));
> > +            Integer distance2 = Math.round(lastKnownPosition.distanceTo(loc2));
> > +
> > +            if (distance1 > distance2) {
>
> En Java on est object, le mieux est de faire de l'object de partout, tu
> peux remplacer ton ton bloc if else par
>
> return distance1.compareTo(distance2)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/24HeuresINSA/24h-android-app/pull/35/files#r26828219>.
>